### PR TITLE
Preferred usage of among/while to amongst/whilst.

### PR DESCRIPTION
**Pull request by smithdc1**
Inspired by @adamchainz's tweet earlier today here is a PR which shows 'while' and 'among' being prefered in the Django docs. 

Wikipeida explains it in more detail but in short:

- In American English, whilst is considered to be pretentious or archaic
- Publications on both sides of the Atlantic disapprove of whilst in their style guides (incl. BBC, Guardian)

https://en.wikipedia.org/wiki/While#Whilst

### CODE REVIEW

File: docs/ref/checks.txt [hunk 1]
```diff
@@ -343,11 +343,11 @@ Models
 * **models.E028**: ``db_table`` ``<db_table>`` is used by multiple models:
   ``<model list>``.
 * **models.E029**: index name ``<index>`` is not unique for model ``<model>``.
-* **models.E030**: index name ``<index>`` is not unique amongst models:
+* **models.E030**: index name ``<index>`` is not unique among models:
   ``<model list>``.
 * **models.E031**: constraint name ``<constraint>`` is not unique for model
   ``<model>``.
-* **models.E032**: constraint name ``<constraint>`` is not unique amongst
+* **models.E032**: constraint name ``<constraint>`` is not unique among
```
Comment (adamchainz) [inline docs/ref/checks.txt:350]: These checks messages changes will also need making in the code. Please can you do a pass checking the whole codebase, not just the docs? There are many user-facing strings in the code.

Comment (smithdc1) [reply docs/ref/checks.txt:350]: Thanks Adam. I've reviewed the codebase and reflected the proposed changes there as well.

**Comment by adamchainz** [comment]
I think it's safest not to backport the code changes just in case someone is making test assertions on the messages.

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: I found one more in tests.

Let's merge this only on master.

**Comment by adamchainz** [comment]
> Let's merge this only on master.

Doesn't that make backporting any future docs changes that touch these lines harder?

**Comment by felixxm** [comment]
@adamchainz Don't worry. We handled backporting with 4a954cfd11a5d034491f87fcbc920eb97a302bb3, sot that's not a big issue :smile:

**Comment by adamchainz** [comment]
👍  That PR is amongst my favourite docs tweaks.

**Pull request merged**